internal/bminventory/inventory.go

```diff
@@ -1668,6 +1676,107 @@ func (b *bareMetalInventory) UpdateClusterNonInteractive(ctx context.Context, pa
+			return nil, nil, nil
+		}
+
+		// case 6
+		if !swag.BoolValue(userManagedNetworking) && (platform == nil || *platform.Type == models.PlatformTypeBaremetal) {
+			if cluster.CPUArchitecture != common.X86CPUArchitecture && !featuresupport.IsFeatureSupported(cluster.OpenshiftVersion, models.FeatureSupportLevelFeaturesItems0FeatureIDARM64ARCHITECTUREWITHCLUSTERMANAGEDNETWORKING) {
+				return nil, nil, common.NewApiError(http.StatusBadRequest, errors.New("disabling User Managed Networking or setting Bare-Metal platform is not allowed for clusters with non-x86_64 CPU architecture"))
+			}
+
+			if *cluster.HighAvailabilityMode == models.ClusterHighAvailabilityModeNone {
+				return nil, nil, common.NewApiError(http.StatusBadRequest, errors.New("disabling User Managed Networking or setting Bare-Metal platform is not allowed in single node mode"))
+			}
+
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeBaremetal)}, swag.Bool(false), nil
+		}
+	}
+
+	return platform, userManagedNetworking, nil
+}
+
+func getActualCreateClusterPlatformParams(platform *models.Platform, userManagedNetworking *bool, highAvailabilityMode *string) (*models.Platform, *bool, error) {
+
+	if err := checkPlatformWrongParamsInput(platform, userManagedNetworking); err != nil {
+		return nil, nil, err
+	}
+
+	if *highAvailabilityMode == models.ClusterHighAvailabilityModeFull {
+		if (platform == nil || *platform.Type == models.PlatformTypeBaremetal) && !swag.BoolValue(userManagedNetworking) {
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeBaremetal)}, swag.Bool(false), nil
+		}
+
+		if swag.BoolValue(userManagedNetworking) || (platform != nil && *platform.Type == models.PlatformTypeNone) {
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}, swag.Bool(true), nil
+		}
+	}
+
+	if *highAvailabilityMode == models.ClusterHighAvailabilityModeNone {
+		if platform != nil && *platform.Type == models.PlatformTypeBaremetal && userManagedNetworking != nil && !*userManagedNetworking {
+			return nil, nil, common.NewApiError(http.StatusBadRequest,
+				errors.Errorf("Can't set %s platform and disable user-managed-networking on single node OpenShift", models.PlatformTypeBaremetal))
+		}
+
+		if platform != nil && *platform.Type == models.PlatformTypeBaremetal {
+			return nil, nil, common.NewApiError(http.StatusBadRequest, errors.Errorf("Can't set %s platform on single node OpenShift", *platform.Type))
+		}
+
+		if userManagedNetworking != nil && !*userManagedNetworking {
+			return nil, nil, common.NewApiError(http.StatusBadRequest, errors.New("Can't disable user-managed-networking on single node OpenShift"))
+		}
+
+		if (userManagedNetworking == nil || *userManagedNetworking) && (platform == nil || *platform.Type == models.PlatformTypeNone) {
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}, swag.Bool(true), nil
+		}
```

Comment:
Don't even need to check it, just return:
```
return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}, swag.Bool(true), nil
```

Reply:
I think that on platforms different than `baremetal` and `none` with `HighAvailabilityMode=None` we still want to return `params.NewClusterParams.Platform` and `params.NewClusterParams.UserManagedNetworking` as they are

Reply:
I added a test to check it for vsphere

Reply:
SNO is always none platform

Reply:
Currently, but what about other platforms? For example I don't think that there is any limitation for SNO + Nutanix

Reply:
@eranco74 is that possible?

Reply:
You can install SNO on Nutanix but the platform will be None!

Reply:
In summary, in case the cluster HighAvailabilityMode is None, any changes to the UMN or platform that result in a platform other than None should return a bad request.

Reply:
Thanks. I'll change it

Reply:
@avishayt @eranco74 I moved the if block that checks if the cluster `HighAvailabilityMode==Full` and `Platform!=None` outside the `platform == nil || *platform.Type == models.PlatformTypeBaremetal` block.
Now the service will always return `BedRequest` when trying to set platform != none on SNO